.github/workflows/publish-image.yml

```diff
@@ -22,13 +22,20 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
-      - name: Build image
-        run: docker build -f Dockerfile -t $IMAGE_NAME .
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
 
-      - name: Login to ghcr.io
-        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
 
-      - name: Push image
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
```

Comment:
This is missing `registry: ghcr.io`, we don't want to log in against Docker Hub 😉.